esmvalcore/preprocessor/_regrid_unstructured.py

```diff
@@ -169,8 +169,11 @@ def _get_weights_and_idx(
         src_points_with_convex_hull = self._add_convex_hull_twice(
             src_points, hull.vertices
         )
-        src_points_with_convex_hull[-2 * n_hull:-n_hull, 1] -= 360
-        src_points_with_convex_hull[-n_hull:, 1] += 360
+        # explicitly casting to int32 since without it, in Numpy 2.0
+        # one gets OverflowError: Python integer 360 out of bounds for int8
+        # see notes https://numpy.org/devdocs/numpy_2_0_migration_guide.html
+        src_points_with_convex_hull[-2 * n_hull:-n_hull, 1] -= np.int32(360)
+        src_points_with_convex_hull[-n_hull:, 1] += np.int32(360)
```

Comment:
Would this work with `360.0` instead of `np.int32(360)`? The points should be floats, not integers, so to me this makes much more sense. If this works, you can also remove the comment

Reply:
not sure, will test - but I agree about points' types indeed

Reply:
nopers, a float there throws this:
```
>       src_points_with_convex_hull[-2 * n_hull:-n_hull, 1] -= 360.
E       numpy._core._exceptions._UFuncOutputCastingError: Cannot cast ufunc 'subtract' output from dtype('float64') to dtype('int8') with casting rule 'same_kind'
```
so we either np.int32 the 360 or we np.float the left member

Reply:
Sorry, I still don't understand this. When I run the tests (numpy < 2), `src_points_with_convex_hull` is an array with dtype `float64` as expected by the CMOR standards. Numpy >= 2 should NOT change that. It would be wrong if this array is of dtype `int`.

I just tested this locally with dummy arrays and it works with `360` and `360.0` (the latter would be "more" correct). Maybe this was just a problem of an early numpy version. Would it be able to test this again?

Reply:
nope, the test [still fails](https://app.circleci.com/pipelines/github/ESMValGroup/ESMValCore/12022/workflows/7aaea67a-5cf4-4978-93e9-890abbca7f1d/jobs/50488?utm_campaign=vcs-integration-link&utm_medium=referral&utm_source=github-checks-link&utm_content=summary) - it's looking at it as `int8` by default, hence my workaround to cast it to `int32`

Reply:
@schlunma directly referencing you since I can't see my comment on the main PR page for some reason (and maybe you can't either, thanks, GitHub)

Reply:
Okay, I found the problem, was looking at the wrong test before. It looks like the default dtype for cubes and their coords is [`int8`](https://github.com/ESMValGroup/ESMValCore/blob/f74a2ada1609b5fa6fa657f6dcda2fee0a029f07/tests/unit/preprocessor/_regrid/__init__.py#L48)...I think it would make sense to change this to something more realistic like `int32` or `int` (`float` might break other tests). There are other tests to that use that, let me try to fix them too. Can I push to this branch?

Reply:
absolutely, please do! :beer: